### PR TITLE
feat[api]: Route for verifying voting key

### LINKED ISSUE #7: [Authentication] Authenticate that a user is eligible to vote and has not already voted.
- [ ] Validate key against database
- [ ] Return signed JWT upon validation

### PR DESCRIPTION
**Pull request by Jatewo**
## Description
This pull request introduces several significant changes, primarily focused on refactoring the test setup, adding new endpoints for key verification and polling stations, and improving environment variable management for GitHub Actions. The most important changes are grouped by theme below.

### Environment and Configuration Updates:
* [`.github/workflows/jest.yml`](diffhunk://#diff-d717659d5699a02ed8f8fe63002776a6697fd148b13640c5faad798a46f9e68dR27-R29): Added environment variables `GITHUB_ACTIONS` and `JWT_SECRET` for GitHub Actions.
* [`config/keys.js`](diffhunk://#diff-79d33c4fac1e062c80d68384f4edd2c2c915c1dae388658f63f02f3f34149f09R5-R17): Updated to conditionally load environment variables based on the presence of `GITHUB_ACTIONS`.

### Test Refactoring:
* [`__test__/controllers/candidates.test.js`](diffhunk://#diff-b25d2a91e1c2fac21ef837e8b0a782a393b2082164d444554fc345c913f5237fL1-L104): Removed old test setup and data population scripts.
* [`__test__/db/populate.js`](diffhunk://#diff-9e1a1c5dcb858e58a1a6054d4fb6df5a3afda9fc7df4bbb9228fac2cca2975c5R1-R26): Added a new script to populate the database with mock data for testing.
* Various test files (`candidateRoutes.test.js`, `constituencyRoutes.test.js`, `districtRoutes.test.js`, `partyRoutes.test.js`): Replaced individual data population scripts with the new `populate` function. [[1]](diffhunk://#diff-f3127855f13f654033977981a0655e825e39ea4732233eb5b4721f414992bfe5L7-R12) [[2]](diffhunk://#diff-14725f6bab1bf2a4957de54440a38cb85320be23d071e9a2d8efd62a1682e518R9) [[3]](diffhunk://#diff-fd7ba966965016b93fca5fdd1b6d7038a512f479a35e469f3ad0e036f398112cL10-R10) [[4]](diffhunk://#diff-cd83675645be48485ba104e0bee8b9809c470b482e2583a0ff9b9c6137e6a357R9)

### New Endpoints:
* [`controllers/keyVerification.js`](diffhunk://#diff-46a50f919e7114643b8f1b4849fa40d99d568d2bcc91af6bdc74c75e3104e526R1-R52): Added a new controller to verify keys and return a JWT token.
* [`controllers/pollingStations.js`](diffhunk://#diff-b8d35b00a30aecdcf7f9d1c9b2667530bd2bb2315efa722f27b40a18efc65c88R1-R36): Added a new controller to get the polling station ID by port number.
* [`routes/index.js`](diffhunk://#diff-103b73514ced7155e7e33849be2ab49fcbf55e6ff018e0e5356a82d74e6c7e09R9-R10): Added routes for key verification and polling stations. [[1]](diffhunk://#diff-103b73514ced7155e7e33849be2ab49fcbf55e6ff018e0e5356a82d74e6c7e09R9-R10) [[2]](diffhunk://#diff-103b73514ced7155e7e33849be2ab49fcbf55e6ff018e0e5356a82d74e6c7e09R21-R22)

### New Tests:
* [`__test__/routes/keyVerificationRoutes.test.js`](diffhunk://#diff-cde5c2821aa90d23cc246a51b98a3f15c62663719a24e80c351b0883431c3fd7R1-R105): Added tests for the key verification endpoint.
* [`__test__/routes/pollingStationRoutes.test.js`](diffhunk://#diff-0fa049b8eab6e49b6c77093040f5e76490a412b34a79cab572bcf16e7f204bc3R1-R58): Added tests for the polling station endpoint.

### Dependency Updates:
* [`package.json`](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R31): Added `jsonwebtoken` dependency and updated Jest configuration to ignore certain paths for coverage. [[1]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R31) [[2]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R47-R52)

### CODE REVIEW

File: __test__/routes/pollingStationRoutes.test.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import request from 'supertest';
+import express from 'express';
+import mongoose from 'mongoose';
+import router from '../../routes/pollingStationRoutes.js'; // Adjust the import path to where your route is located
```
Comment (MortenMunk) [inline __test__/routes/pollingStationRoutes.test.js]: Is this comment a remnant from GPT? :rofl:

**Review by MortenMunk — CHANGES_REQUESTED** [review]
lgtm just a minor comment

**Review by TheMOHD3886 — DISMISSED** [review]
lgtm 👍

**Review by MortenMunk — APPROVED** [review]
lgtm

**Review by TheMOHD3886 — APPROVED** [review]
lgtm again 👍

**Pull request merged**